src/aws/cloudwatch-sink.cpp

```diff
@@ -420,11 +435,19 @@ void CloudWatchSink::restart_sink() noexcept
         report_logger_error("Requested to restart sink but sink is not running!", log_group_name_, "");
         return;
     }
-    stop_impl();
-    is_running_ = true;
-    thread_pid_ = ::getpid();
-    aws_cloudwatch_client_ = Aws::MakeUnique<Aws::CloudWatchLogs::CloudWatchLogsClient>("cloudwatch");
-    cloudwatch_logs_thread_ = std::make_unique<std::thread>(&CloudWatchSink::cloudwatch_logs_thread, this);
+    try
+    {
+        stop_impl();
+        logs_queue_.clear();
+        is_running_ = true;
+        thread_pid_ = ::getpid();
+        aws_cloudwatch_client_ = Aws::MakeUnique<Aws::CloudWatchLogs::CloudWatchLogsClient>("cloudwatch");
+        cloudwatch_logs_thread_ = std::make_unique<std::thread>(&CloudWatchSink::cloudwatch_logs_thread, this);
+    }
+    catch (const std::exception& e)
+    {
+        // Ignored, just so the thread itself will not die
```

Comment:
Here and in line 427, it's unclear to me why the exceptions are ignored. 
Should we expect an exception in standard flows? If not, why not at least log the error?

Reply:
because we are talking about logging, and those places are critical on forking, we do not want to print anything, to not cause a lock on fork issue, either from stdout or from our logging